src/TTextEdit.cpp

```diff
@@ -1566,8 +1566,12 @@ QString TTextEdit::getSelectedText(char newlineChar)
         // if the selection started on this line
         if (y == mPA.y()) {
             // start from the column where the selection started
-            if (!mpBuffer->buffer.at(y).at(0).isSelected()) {
-                x = mPA.x();
+            if (mpBuffer->buffer.size() > y) { //Check if the index will be in range first
+                if (mpBuffer->buffer.at(y).size() >= 0) { //Check if the index will be in range first
```

Comment:
`TBuffer::buffer` is a `std::deque<std::deque<TChar>>` as such, if `y < mpBuffer->buffer.size()` then `mpBuffer->buffer.at(y)` is EITHER `empty()` (corresponding to `mpBuffer->buffer.at(y).size()` being `0`) OR not (corresponding to a value *greater* than zero). Indeed `mpBuffer->buffer.at(y).size()` returns a [`std::size_t`](https://en.cppreference.com/w/cpp/types/size_t) type of value - which is an *unsigned* integer type so the test for it being `>= 0` is a pointless test ...

tl;dr; This test is redundant IMHO.

Reply:
I agree, thank you. That means the crash bug must have been on the other line since there were only 2 places I could see any crash bug being.